Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceRayTracingPipelineState.h

```diff
@@ -0,0 +1,127 @@
+
+        const RayTracingShaderLibraryVector& GetShaderLibraries() const
+        {
+            return m_descriptor.GetShaderLibraries();
+        }
+        RayTracingShaderLibraryVector& GetShaderLibraries()
+        {
+            return m_descriptor.GetShaderLibraries();
+        }
+
+        const RayTracingHitGroupVector& GetHitGroups() const
+        {
+            return m_descriptor.GetHitGroups();
+        }
+        RayTracingHitGroupVector& GetHitGroups()
+        {
+            return m_descriptor.GetHitGroups();
+        }
+
+        //! Build operations
+        MultiDeviceRayTracingPipelineStateDescriptor* Build();
+        MultiDeviceRayTracingPipelineStateDescriptor* MaxPayloadSize(uint32_t maxPayloadSize);
+        MultiDeviceRayTracingPipelineStateDescriptor* MaxAttributeSize(uint32_t maxAttributeSize);
+        MultiDeviceRayTracingPipelineStateDescriptor* MaxRecursionDepth(uint32_t maxRecursionDepth);
+        MultiDeviceRayTracingPipelineStateDescriptor* PipelineState(const RHI::MultiDevicePipelineState* pipelineState);
+        MultiDeviceRayTracingPipelineStateDescriptor* ShaderLibrary(RHI::PipelineStateDescriptorForRayTracing& descriptor);
+
+        MultiDeviceRayTracingPipelineStateDescriptor* RayGenerationShaderName(const AZ::Name& name);
+        MultiDeviceRayTracingPipelineStateDescriptor* MissShaderName(const AZ::Name& name);
+        MultiDeviceRayTracingPipelineStateDescriptor* ClosestHitShaderName(const AZ::Name& closestHitShaderName);
+        MultiDeviceRayTracingPipelineStateDescriptor* AnyHitShaderName(const AZ::Name& anyHitShaderName);
+
+        MultiDeviceRayTracingPipelineStateDescriptor* HitGroup(const AZ::Name& name);
+
+    private:
+        const RHI::MultiDevicePipelineState* m_mdPipelineState = nullptr;
+        RayTracingPipelineStateDescriptor m_descriptor;
+    };
+
+    //! Defines the shaders, hit groups, and other parameters required for ray tracing operations across multiple devices.
+    class MultiDeviceRayTracingPipelineState : public MultiDeviceObject
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(MultiDeviceRayTracingPipelineState, AZ::SystemAllocator, 0);
+        AZ_RTTI(MultiDeviceRayTracingPipelineState, "{22F609DF-C889-4278-9580-3D2A99E78857}", MultiDeviceObject);
+        AZ_RHI_MULTI_DEVICE_OBJECT_GETTER(RayTracingPipelineState);
+        MultiDeviceRayTracingPipelineState() = default;
+
+        virtual ~MultiDeviceRayTracingPipelineState() = default;
+
+        const MultiDeviceRayTracingPipelineStateDescriptor& GetDescriptor() const
+        {
+            return m_mdDescriptor;
+        }
+
+        //! Initialize all device-specific RayTracingPipelineStates
+        ResultCode Init(MultiDevice::DeviceMask deviceMask, const MultiDeviceRayTracingPipelineStateDescriptor& descriptor);
+
+    private:
+        //! explicit shutdown is not allowed for this type
+        void Shutdown() override final;
+
+        MultiDeviceRayTracingPipelineStateDescriptor m_mdDescriptor;
```

Comment:
Is the only reason to create a MD version of the descriptor is because the descriptor holds a PSO that needs to be a MD PSO?

Reply:
Pretty much. It is there to have the API not change for RPI level or higher up users as much as possible.

(Warning: rant incoming, there is probably no point in reading this:)
We dislike the RT code quite a bit because the design is completely different from all the other parts of Atom, i.e., there is no consistency in the API design and all the RT code feels like an out of place attachment to Atom rather than a properly integrated part. This is not only a Frankenstein's monster in the RHI - it gets even worse when you look higher up and consider stuff like the integration with materials, which is just not there. I don't want to blame anyone here, but the RT code just feels incomplete, not well thought through and rushed into the engine just to have the feature checked off. /rant

Reply:
Integration with material -> I had a chat with someone from your team in the last sig/graphics meeting about considering access to materials via Bindless which will open up the potential to have any pipeline access any material properties. The idea is that we want to have forward , deferred, RT pipeline, etc all able to use bindless to access buffers that contain material data and use them however they want. 

We would have moved to such a system sooner except the mobile held us back. Unbounded Arrays are still not supported on 5 yr old devices and until we can up the min spec by 1-2 years we will have to keep the current way of binding materials alive. But that's not to say we can't switch to the bindless system for hw that does support unbounded arrays.